scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -817,19 +923,28 @@ def run(self):
             if retries == 0 or (self.results.failed == self.results.error and not self.options.retry_build_errors):
                 break
 
-    def update_counting(self):
```

Comment:
The stats logic also needs to be adjusted because the zephyr.symbols is
generated after build. So ExecutionCounter update is split and some
must be postponed until test execution pipeline is completed.